### PR TITLE
build: get binary build info from `debug/buildinfo`

### PR DESCRIPTION
**Pull request by positiveblue**
By using the `debug/buildinfo` package we do not have to relay on setting custom ldflags.
    
Unfortunately, there is no `vcs.tags` settings so we need to still pass the output of `git describe --tags --dirty` as an ldflag.

Note: this change was already merged in [LND](https://github.com/lightningnetwork/lnd/pull/6963) and the lnd package need to be updated in `go.mod` for this to work.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -96,12 +88,12 @@ build-itest:
...
 
 	@$(call print, "Building itest lnd.")
-	CGO_ENABLED=0 $(GOBUILD) -tags="$(ITEST_TAGS)" -o itest/lnd-itest $(ITEST_LDFLAGS) $(LND_PKG)/cmd/lnd
+	CGO_ENABLED=0 $(GOBUILD) -tags="$(ITEST_TAGS)" -o itest/lnd-itest $(DEV_LDFLAGS) $(LND_PKG)/cmd/lnd
 
 install:
 	@$(call print, "Installing tarod and tarocli.")
-	$(GOINSTALL) -tags="${tags}" $(LDFLAGS) $(PKG)/cmd/tarod
-	$(GOINSTALL) -tags="${tags}" $(LDFLAGS) $(PKG)/cmd/tarocli
+	$(GOINSTALL) -tags="${tags}" $(RELEASE_LDFLAGS) $(PKG)/cmd/tarod
```
Comment (bhandras) [inline Makefile:95]: Two things, when I run `make install` I get:
```
$ make install
fatal: No names found, cannot describe anything.
 Installing tarod and tarocli.
GO111MODULE=on go install -v -tags=""  -s -w -buildid= -X github.com/lightninglabs/taro/build.Commit= github.com/lightninglabs/taro/cmd/tarod
flag provided but not defined: -s
usage: go install [build flags] [packages]
Run 'go help install' for details.
make: *** [Makefile:95: install] Error 2
```

And also when I just `make` the `./tarod-debug --version` command doesn't anymore have the commit set.

Comment (positiveblue) [reply Makefile:95]: you were right, I forgot the `-ldflags` in front of  `$(RELEASE_LDFLAGS)` same [here](https://github.com/lightningnetwork/lnd/pull/7050)

Thank you for noticing @bhandras :pray:

**Comment by positiveblue** [comment]
I updated lnd to ensure that it had the latest changes in master. However, I am not sure if there is anything that we do not want to get in the project yet :thinking:

**Review by bhandras — APPROVED** [review]
tACK! :tada:

**Pull request merged**